packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js

```diff
@@ -54,6 +57,19 @@ const Columns = ({
     },
     [columns, setColumnsObject]
   );
+
+  const setSelectAllCheckBox = () => {
+    let checkboxes = document.querySelectorAll(
+      `.${blockClass}__customize-columns-modal .bx--checkbox`
+    );
+    let selectAll = getVisibleColumnsCount() != columns.length;
+
+    for (let i = 1; i < checkboxes.length; i++) {
+      checkboxes[i].checked = selectAll;
+      onSelectColumn(columns[i - 1], selectAll);
+    }
+  };
+
```

Comment:
Instead of setting the `checked` attributes explicitly, update the `columns` array. Once the `columns` array is updated, React will re-render the checkboxes with the updated state.

The following code from `CustomizeColumnsModal.js` may be a useful reference. Take a look and see a similar technique can be applied here:
https://github.com/carbon-design-system/ibm-cloud-cognitive/blob/94f3ba91fefcc8acdcb8dc12d3db6d2287e0bc81/packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/CustomizeColumnsModal.js#L65

Reply:
Removed `checkboxes[i].checked = selectAll;` since it is redundant